File: 2025/day1/part2.py
input_data = []
try:
    with open("input.txt") as file:
        for line in file:
            line = line.strip()
            if line:
                input_data.append(line)
except FileNotFoundError:
    with open("2025/day1/input.txt") as file:
        for line in file:
            line = line.strip()
            if line:
                input_data.append(line)


def main(input_data):
    current = 50
    zeros = 0

    for line in input_data:
        direction, clicks = line[0], int(line[1:])

        for _ in range(clicks):
            if direction == "R":
                current = (current + 1) % 100
            else:
                current = (current - 1) % 100
                if current < 0:
                    current = (current + 100) % 100

            if current == 0:
                zeros += 1

    return zeros


result = main(input_data)
print(result)

File: 2025/day1/part2.js
const fs = require("node:fs");
let input = "";
try {
    const data = fs.readFileSync("input.txt", "utf8");
    if (data) input += data;
} catch {
    const data = fs.readFileSync("2025/day1/input.txt", "utf8");
    if (data) input += data;
}
input = input.split("\n");

function main(input_data) {
    let current = 50;
    let zeros = 0;

    for (line of input_data) {
        const direction = line.slice(0, 1);
        const clicks = Number.parseInt(line.slice(1));

        for (let i = clicks; i > 0; i--) {
            if (direction === "R") {
                current = (current + 1) % 100;
            } else {
                current = (current - 1) % 100;
                if (current < 0) {
                    current = (current + 100) % 100;
                }
            }

            if (current === 0) {
                zeros += 1;
            }
        }
    }
    return zeros;
}

const result = main(input);
console.log(result);
